torch/_dynamo/symbolic_convert.py

```diff
@@ -581,6 +581,33 @@ def handle_graph_break(
     return decorator
 
 
+def promote_exception_to_graph_break(inner_fn):
+    @functools.wraps(inner_fn)
+    def wrapper(self: "InstructionTranslatorBase", inst: Instruction):
+        try:
+            TracingContext.set_current_loc(
+                self.f_code.co_filename, self.lineno, self.f_code.co_name
+            )
+            return inner_fn(self, inst)
+        except Unsupported:
+            raise
+        except RuntimeError as e:
+            if inst.offset is None or not hasattr(self.f_code, "co_exceptiontable"):
+                raise
+
+            from torch._dynamo.bytecode_transformation import parse_exception_table
+
+            table = parse_exception_table(self.f_code.co_exceptiontable)
+            for entry in table:
+                if entry.start <= inst.offset < entry.end:
+                    # If we are in a try block of a try-catch, promote the
+                    # exception to graph break
+                    raise Unsupported(str(e)) from e
```

Comment:
I want a sanity check from the Dynamo folks (@anijain2305, @yanboliang) on this. If an exception gets thrown inside of a try-catch block, we end up promoting it to a graph break (instead of allowing it through). I hope this is not another "suppress_errors=True" situation; the argument is that (1) most people won't throw their code under try-catch and (2) we should only try to convert errors that are not internal dynamo errors into graph breaks

Reply:
If we are worried about a suppress_errors=True situation, then maybe we could limit this change to just errors being thrown from FakeTensor prop?